### PR TITLE
Add condition on test method

### PR DESCRIPTION
**Pull request by zguesmi**
This fixes:
```
toomanyrequests: too many failed login attempts for username or IP address.
```

### CODE REVIEW

**Review by mcornaton — APPROVED** [review]

File: src/test/java/com/iexec/common/docker/client/EnabledOnlyOnWednesdayNightCondition.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+        boolean isWednesdayNight = LocalDateTime.now().getDayOfWeek().equals(DayOfWeek.WEDNESDAY) // wednesday == 3
+                && LocalDateTime.now().getHour() >= 18; // 6pm
+        if (!isWednesdayNight) {
+            return ConditionEvaluationResult.disabled("Runs only on wednesday in the night");
+        }
+        return ConditionEvaluationResult.enabled("It's wednesday!");
+    }
+
+    public static void main(String[] args) {
```
Comment (james-toussaint) [inline src/test/java/com/iexec/common/docker/client/EnabledOnlyOnWednesdayNightCondition.java]: To remove?

Comment (zguesmi) [reply src/test/java/com/iexec/common/docker/client/EnabledOnlyOnWednesdayNightCondition.java]: Oups!

Comment (zguesmi) [reply src/test/java/com/iexec/common/docker/client/EnabledOnlyOnWednesdayNightCondition.java]: Thanks for the reminder.

Comment (zguesmi) [reply src/test/java/com/iexec/common/docker/client/EnabledOnlyOnWednesdayNightCondition.java]: Done.

**Comment by jbern0rd** [comment]
It works but this is lots of code + a main method for a specific use case.
Why not only use @EnableIf with a method implementing the check ?
https://junit.org/junit5/docs/current/user-guide/#writing-tests-conditional-execution-custom

**Comment by zguesmi** [comment]
I thought `@EnableIf` was deprecated starting from Junit 5.5 ([ref](https://www.baeldung.com/junit-5-conditional-test-execution#1-deprecation-notice)). I didn't find it in our current version.

**Review by james-toussaint — APPROVED** [review]
Nice catch!

**Comment by zguesmi** [comment]
I think the simplest thing to do is to disable the test for now. I agree with @jbern0rd .

File: src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java [hunk 2]
```diff
@@ -163,7 +163,14 @@ public void shouldGetAuthenticatedClientWithDockerIoRegistry() throws Exception
                 .isEqualTo(dockerIoPassword);
     }
 
+    /**
+     * This test is temporarily disabled because of this error:
+     * toomanyrequests: too many failed login attempts for
+     * username or IP address.
+     * It runs only on wednesday night.
```
Comment (james-toussaint) [inline src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java]:
```suggestion
```

**Review by james-toussaint — APPROVED** [review]

File: src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java (hunk 2 above)
Comment (zguesmi) [reply src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java]: Thanks.

**Review by jbern0rd — APPROVED** [review]

**Comment by jbern0rd** [comment]
> I thought `@EnableIf` was deprecated starting from Junit 5.5 ([ref](https://www.baeldung.com/junit-5-conditional-test-execution#1-deprecation-notice)). I didn't find it in our current version.

I missed this, thank you for the notice.

**Comment by jbern0rd** [comment]
If we want to keep such a test, we could fail the authentication on another registry too

**Comment by zguesmi** [comment]
> > I thought `@EnableIf` was deprecated starting from Junit 5.5 ([ref](https://www.baeldung.com/junit-5-conditional-test-execution#1-deprecation-notice)). I didn't find it in our current version.
> 
> I missed this, thank you for the notice.

I think it is back in version 5.7

**Pull request merged**